axelor-contract/src/main/java/com/axelor/apps/contract/batch/BatchContractRevaluate.java

```diff
@@ -0,0 +1,218 @@
+    List<Contract> contractList;
+    while (!(contractList =
+            query
+                .bind("onGoingStatus", ContractVersionRepository.ONGOING_VERSION)
+                .bind("idsOk", idsOk)
+                .bind("idsFail", idsFail)
+                .fetch(FETCH_LIMIT))
+        .isEmpty()) {
+      Map<String, List<Contract>> ids = revaluateContracts(contractList);
+      idsOk.addAll(ids.get("OK").stream().map(Contract::getId).collect(Collectors.toList()));
+      idsFail.addAll(ids.get("FAIL").stream().map(Contract::getId).collect(Collectors.toList()));
+      idsReevaluated.addAll(
+          ids.get("REEVALUATED").stream().map(Contract::getId).collect(Collectors.toList()));
+      JPA.clear();
+    }
+    LOG.debug("{} Reevaluated contracts : {}", idsReevaluated.size(), idsReevaluated);
+  }
+
+  protected Map<String, List<Contract>> revaluateContracts(List<Contract> contractList) {
+    List<Contract> idsFail = new ArrayList<>();
+    List<Contract> idsOk = new ArrayList<>();
+    List<Contract> idsReevaluated = new ArrayList<>();
+    for (Contract contract : contractList) {
+      processContract(idsFail, idsOk, idsReevaluated, contract);
+    }
+    return Stream.of(
+            new AbstractMap.SimpleEntry<>("OK", idsOk),
+            new AbstractMap.SimpleEntry<>("FAIL", idsFail),
+            new AbstractMap.SimpleEntry<>("REEVALUATED", idsReevaluated))
+        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+  }
+
+  @Transactional(rollbackOn = Exception.class)
+  protected void processContract(
+      List<Contract> idsFail,
+      List<Contract> idsOk,
+      List<Contract> idsReevaluated,
+      Contract contract) {
+    try {
+      contract = contractRepository.find(contract.getId());
+      Contract newContract = contractService.getNextContract(contract);
+      processContract(newContract, idsOk, idsReevaluated);
+    } catch (Exception e) {
+      TraceBackService.trace(e, null, batch.getId());
+      idsFail.add(contract);
+      incrementAnomaly();
+    }
+  }
+
+  void processContract(Contract contract, List<Contract> idsOk, List<Contract> idsReevaluated)
+      throws AxelorException, ScriptException {
+    LocalDate todayDate = appBaseService.getTodayDate(contract.getCompany());
+    LocalDate nextRevaluationDate = contract.getNextRevaluationDate();
+    LocalDate lastRevaluationDate = contract.getLastRevaluationDate();
+
+    if (launchBatch(todayDate, contract)) {
+      contract = contractRevaluationService.applyFormula(contract);
+      contract.setLastRevaluationDate(todayDate);
+      contract.setNextRevaluationDate(computeNextRevaluationDate(contract, todayDate));
+
+      idsReevaluated.add(contract);
+      incrementDone();
+    } else if (nextRevaluationDate == null && lastRevaluationDate != null) {
+      contract.setNextRevaluationDate(
+          computeNextRevaluationDate(contract, lastRevaluationDate).minusDays(5));
```

Comment:
I'm not sure why it is precisely 5 days, but maybe it would be interesting to define a variable or an attribute for this constant (later it could become a config)